### PR TITLE
Add support for skills

### PR DESCRIPTION
**Pull request by mrubens**
Per https://agentskills.io/

----

> [!IMPORTANT]
> Adds support for skills management and integration into system prompts, with a new `SkillsManager` class and comprehensive tests.
> 
>   - **Behavior**:
>     - Adds `SkillsManager` class in `SkillsManager.ts` to manage skills, including discovery and file watching.
>     - Integrates skills into system prompt generation in `generateSystemPrompt.ts` by adding `getSkillsSection()`.
>     - Updates `ClineProvider.ts` to initialize and manage `SkillsManager`.
>   - **Skills Management**:
>     - Supports global and project-specific skills, with mode-specific overrides.
>     - Uses `SKILL.md` files for skill metadata and content.
>     - Provides functions to get skills for a specific mode and full skill content.
>   - **Testing**:
>     - Adds tests for `SkillsManager` in `SkillsManager.spec.ts` to verify skill discovery, mode filtering, and content retrieval.
>     - Mocks file system and VSCode APIs for testing purposes.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 1f9e4720f59c26d042d329689dceee4dcb231ba9. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by ghost** [comment]
> **Rooviewer**  Clock &nbsp;&nbsp;[See task on Roo Cloud](https://app.roocode.com/cloud-jobs/3rpAF6XD?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review.sync)

Review complete. All previous issues have been resolved.

This PR implements a well-structured skills system based on the Agent Skills specification (https://agentskills.io/). The implementation includes:

- **SkillsManager service** - Discovers and manages skills from global (`~/.roo/skills`) and project (`.roo/skills`) directories with proper symlink support
- **Mode-specific skills** - Supports `skills-{mode}/` directories for mode-specific skill overrides
- **Override resolution** - Correctly prioritizes project over global, and mode-specific over generic skills
- **System prompt integration** - Skills section is properly injected into the system prompt for the current mode
- **Lifecycle management** - Proper initialization, file watching, and disposal with WeakRef to prevent memory leaks
- **Comprehensive tests** - Good test coverage including edge cases for symlinks, missing fields, and override resolution

- [x] Avoid circular dependency in `SkillsManager` by switching `ClineProvider` import to type-only (`import type`)
- [x] Avoid leaking absolute local filesystem paths in the system prompt skills list (show workspace-relative path)

> <details>
> <summary>Previous reviews</summary>
> 
> - a23977d: [Review #1](https://app.roocode.com/cloud-jobs/71DO9Dsm?utm_source=previous-reviews&utm_medium=link&utm_campaign=github.pr.review)
> - a23977d: [Review #2](https://app.roocode.com/cloud-jobs/z2nKrBP7?utm_source=previous-reviews&utm_medium=link&utm_campaign=github.pr.review)
> - 38be044: [Review #3](https://app.roocode.com/cloud-jobs/pFPfZKVw?utm_source=previous-reviews&utm_medium=link&utm_campaign=github.pr.review)
> - 0e4c71b: [Review #4](https://app.roocode.com/cloud-jobs/e01BfIvt?utm_source=previous-reviews&utm_medium=link&utm_campaign=github.pr.review)
> - 4c6c26c: [Review #5](https://app.roocode.com/cloud-jobs/Lh8HOCiy?utm_source=previous-reviews&utm_medium=link&utm_campaign=github.pr.review)
> </details>
<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=RooCodeInc/Roo-Code&prNumber=10335&commentId=3690981507).</sub>

**Comment by ghost** [comment]
> **Oroocle**  Clock &nbsp;&nbsp;[Follow along on Roo Cloud](https://app.roocode.com/cloud-jobs/71DO9Dsm?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review)

Review in progress. Left a couple of actionable notes.

- [ ] Avoid circular dependency in `SkillsManager` by switching `ClineProvider` import to type-only (`import type`).

- [ ] Avoid leaking absolute local filesystem paths in the system prompt skills list (show workspace-relative path or omit path).

<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=RooCodeInc/Roo-Code&prNumber=10335&commentId=3691027683).</sub>

File: src/services/skills/SkillsManager.ts [hunk 1]
```diff
@@ -0,0 +1,329 @@
+import * as fs from "fs/promises"
+import * as path from "path"
+import * as vscode from "vscode"
+import matter from "gray-matter"
+
+import { ClineProvider } from "../../core/webview/ClineProvider"
```
Comment (ghost) [inline src/services/skills/SkillsManager.ts]: `SkillsManager.ts` imports `ClineProvider` as a runtime value, creating a circular dependency (`ClineProvider` -> `SkillsManager` -> `ClineProvider`). Since this is only used for types, switch this import to `import type` to avoid potential partial initialization issues.

```suggestion
import type { ClineProvider } from "../../core/webview/ClineProvider"
```

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=10335&fixId=cad68058f1) or mention @roomote and request a fix.</sub>

Comment (ghost) [inline src/services/skills/SkillsManager.ts]: Follow-up to my prior inline suggestion: the exact type-only import line should be:

```suggestion
import type { ClineProvider } from "../../core/webview/ClineProvider"
```

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=10335&fixId=eea72c5f87) or mention @roomote and request a fix.</sub>

File: src/core/prompts/sections/skills.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	// Separate generic and mode-specific skills for display
+	const genericSkills = skills.filter((s) => !s.mode)
+	const modeSpecificSkills = skills.filter((s) => s.mode === currentMode)
+
+	let skillsList = ""
+
+	if (modeSpecificSkills.length > 0) {
+		skillsList += modeSpecificSkills
+			.map((skill) => `  * "${skill.name}" skill (${currentMode} mode) - ${skill.description} [${skill.path}]`)
```
Comment (ghost) [inline src/core/prompts/sections/skills.ts]: `getSkillsSection()` prints absolute filesystem paths in the system prompt (via `skill.path`). Since the system prompt is sent to a remote model, this can leak local machine usernames/home dirs or other sensitive layout details. Consider displaying a workspace-relative path, or omitting the path entirely and only showing the skill name, then resolve the actual file path internally.

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=10335&fixId=db22d7c045) or mention @roomote and request a fix.</sub>

Comment (mrubens) [reply src/core/prompts/sections/skills.ts]: Yeah makes sense, should be relative paths

**Comment by mrubens** [comment]
@roomote please fix the issues flagged in the comments

**Comment by xiaoqiangch** [comment]
hey, guys, how to use this staff, looks cool, I have get a plugin that will load skills to a folder call .github under the project folder. I don't think that will work,right? any ideas how to fix it. are there any settings? thanks a lot!

**Comment by mrubens** [comment]
> hey, guys, how to use this staff, looks cool, I have get a plugin that will load skills to a folder call .github under the project folder. I don't think that will work,right? any ideas how to fix it. are there any settings? thanks a lot!

It will load skills from .roo/skills (which can be a symlink to your .github folder if you want). Let me know if that works.

**Comment by HappyVing** [comment]
Can the roo page support the import of skill functions, for example:

image
